streams/src/test/java/org/apache/kafka/streams/state/internals/CachingInMemoryKeyValueStoreTest.java

```diff
@@ -359,6 +361,34 @@ public void shouldReverseIterateOverRange() {
         ), results);
     }
 
+    @Test
+    public void shouldGetRecordsWithPrefixKey() {
```

Comment:
In all the tests for the prefix scan you should also verify boundary conditions, e.g., if you have a prefix `abcd`, you should verify that `abce` is not matched since this is the first key that should not be matched.

Reply:
i have added wherever the boundary conditions were not getting tested.